enhancements

thiswillbeyourgithub opened the pull request:
Hi, I took the liberty of adding some enhancements. I hope that's okay with you.

Here's a breakdown of each commit:

1. Use best cheapest model
**fix: use cheapest best model**

2. Instead of json formatting, use n parameter
- **fix: don't ask the LLM to output json, instead use multiple generations and deduplicate them**

3. Add env variable for the llm name
- **make the llm name easily modifiable**
- **make it simple to change the number of generations to do**

4. don't run if the prompt is empty
- **fix: don't run if no prompt**

5. Add an eplainer mode, where the LLM outputs commented oneliners
- **feat: add an explainer mode**

6. misc
- **missing newline**
- **fix: forgot quotation sign**
- **clearer workflow**
- **better error message**
- **fix llm name in print message**
- **set default completions to 5**
- **update readme**


edit: also the LLM now knows which shell is used

thiswillbeyourgithub committed:
fix: use cheapest best model

thiswillbeyourgithub committed:
fix: don't ask the LLM to output json, instead use multiple generations and deduplicate them

thiswillbeyourgithub committed:
make the llm name easily modifyable

thiswillbeyourgithub committed:
missing newline

thiswillbeyourgithub committed:
fix: firgot quotation sign

thiswillbeyourgithub committed:
feat: make it simple to change the number of generations to do

thiswillbeyourgithub committed:
clearer workflow

thiswillbeyourgithub committed:
fix: don't run if no prompt

thiswillbeyourgithub committed:
feat: add an explainer mode

thiswillbeyourgithub committed:
better error message

thiswillbeyourgithub committed:
fix llm name in print message

thiswillbeyourgithub committed:
set default completions to 5

thiswillbeyourgithub committed:
update readme

devnoname120 commented on zsh-ai-commands.zsh:
```diff
@@ -3,50 +3,97 @@
+  ZSH_AI_COMMANDS_SUGGESTIONS=$(echo $ZSH_AI_COMMANDS_GPT_RESPONSE | jq -r '.choices[].message.content' | uniq)
+fi
+
+  ZSH_AI_COMMANDS_SELECTED=$(echo $ZSH_AI_COMMANDS_SUGGESTIONS | fzf)
```
`fzf` currently opens in fullscreen mode which is pretty annoying. This fixes it so that it appears inline:

```suggestion
  ZSH_AI_COMMANDS_SELECTED=$(echo $ZSH_AI_COMMANDS_SUGGESTIONS | fzf --height=~100%)
```

thiswillbeyourgithub committed:
better fzf with previews

thiswillbeyourgithub committed:
fix: only use preview of fzf if using explainer

thiswillbeyourgithub committed:
forgot to wrap

thiswillbeyourgithub committed:
fix: proper height for fzf with a recent fzf version

thiswillbeyourgithub committed:
feat: show the explanation in the preview, but not in the command

thiswillbeyourgithub committed:
a desperate parser attempt

muePatrick committed:
Update Readme

- added a hint that the apt version of fzf is too outdated
- updated the gif to preview the explain feature

muePatrick commented:
Hey,
the enhancements are absolutely okay with me :smile:.
Thank you so much for your awesome contribution, I am sorry it took my so long to look into them.

I update the Readme to showcase your new preview feature. And also added a note about the fzf version (had to find out the hard way how old the apt version is :grin:).

thiswillbeyourgithub commented:
Nice!

>  update the Readme to showcase your new preview feature.

Thanks!

> And also added a note about the fzf version (had to find out the hard way how old the apt version is 😁).

Same here, thanks!

As you can see in my latest commit I added something a bit weird : actually when the llm answer contains some terminal codes the jq parsing often fails miserably. So I added some ugly code that aggressively removes some things frequently found in terminal codes like color codes etc. But it was such a pain to make this work using sed that I ended up using [pypipe](https://github.com/bugen/pypipe) to get it working quickly.

So
1 using pypipe like that is ugly, if anyone finds a way to use sed that's better
2 in effect I added a pypipe dependency, you could mention it in the readme but I would prefer getting rid of this if someone makes it work with sed

thiswillbeyourgithub commented:
I made a new PR, no more need for pypipe wizardry
